### DISCUSSION TITLE
how can i remove the default " Do you want to save the changes to the following 1 files?" dialog?

### DISCUSSION THREAD

**Discussion by Keithyumi** [eclipse-theia/theia#14163] [Q&A] [answered]
<img width="422" alt="image" src="https://github.com/user-attachments/assets/5fe51e4e-4abb-4cf1-bb4c-cd8f337e6401">

Hello, I'm looking to develop my own IDE based on Theia and Electron. 

When I create a  electron window and open a project using "WorkspaceService.open" command and then there are unsaved files when I attempt to close the electron window, clicking the 'don't save' button doesn't close window properly. Only clicking the 'Save All' button works. 

How can I define the events that occur after clicking the 'Discard Changes' button or disable the default blocking dialog that prevents closing? thanks.

**Comment by msujew** [comment]
Hey @Keithyumi,

which version of Theia are you using? I believe what you're experiencing is a bug that was fixed in 1.53 via https://github.com/eclipse-theia/theia/pull/14062.

Aside from that, you can change the save behavior by overriding the `FileSystemSaveableService` and rebind it as the `SaveableService` in your DI container.

**Reply by Keithyumi** [reply]
my theia-cli version is 1.43.0, I attempt to install 1.53.0 but Couldn't find any versions for "@theia/cli" that matches "1.53.0"

**Reply by msujew** [reply]
Ok, I have no clue why this doesn't work for you. The versions are definitely available on npm. I would recommend to install the latest patch release (1.53.2):

![image](https://github.com/user-attachments/assets/dd0f62e6-2f2c-441b-a066-a6551efc6d54)

**Reply by Keithyumi** [reply]
ok, I have updated the theia to 1.53.2, but
it will conflict with my current dependencies, how can I solve the problem in 1.43.0

**Reply by msujew** [reply]
I don't think it's possible to change the behavior easily. The code for that lives inside of a function and not in a service.

It was only recently refactored into a service see https://github.com/eclipse-theia/theia/pull/13683 that was released with Theia 1.50. You will need to dig into the 1.43 source code to figure something out.

1.43 is an unsupported version of Theia.

**Reply by Keithyumi** [reply]
It's my problem, when i  close tab and click don't save button ,the tabs closed properly, but close the window and click the button doesn't works. 

It caused by 1.43.0 version?

**Reply by msujew** [reply, marked as answer]
Can you retry this with the Electron example app in this repo? I cannot really help with unsupported versions, see also the reasons I gave in https://github.com/eclipse-theia/theia/discussions/14163#discussioncomment-10645989.

**Reply by Keithyumi** [reply]
ok, i confirm that the bug is relevant with the version via theia-blueprint 1.43.0,  now i will attempt to update my theia version. thanks vvvvvvv very much!
